src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -135,24 +131,19 @@ public MesosCloud(
 
     this.agentUser = agentUser;
     this.role = role;
-    this.mesosAgentSpecTemplates = mesosAgentSpecTemplates;
+    this.mesosAgentSpecTemplates =
+        mesosAgentSpecTemplates != null ? mesosAgentSpecTemplates : Collections.emptyList();
```

Comment:
Hudson.util has an static method for fixing empty collections
https://github.com/jenkinsci/jenkins/blob/598d78b8b570b6eefffbc75b7df6dc0d29220232/core/src/main/java/hudson/Util.java#L1065-L1119
```suggestion
    this.mesosAgentSpecTemplates = Util.fixNull(mesosAgentSpecTemplates);
```